src/go/languageservice/workspace.go

```diff
@@ -169,20 +171,54 @@ func (id WorkspaceID) GetWorkspaceErrors() []error {
 	return errors
 }
 
-func (id WorkspaceID) GetCompletions() ([]string, error) {
+func (id WorkspaceID) GetCompletions(position int) ([]string, error) {
 
 	workspace, err := id.getWorkspace()
 	if err != nil {
 		return nil, *err
 	}
 
 	completions := []string(nil)
-
-	// TODO: take into account context.
-	// TODO: support locals.
+    
+	// TODO: support other node types.
 	// TODO: return item type.
 	// TODO: support fetching item description.
 	for _, wd := range workspace.Files {
+        codePos := token.Pos(position)
+        // current position: =  wd.FileSet.Position(codePos).String()
+        enclosingPath, _ := astutil.PathEnclosingInterval(wd.File, codePos, codePos+1)
+	    
+        // locals form enslosing scopes, currently only function declarations:
+	    for _ , enclosingNode := range(enclosingPath) {
+		    switch x := enclosingNode.(type) {
+
+                // for function declarations, get all their locals declared before codePos 
+		        case *ast.FuncDecl:
+			        for _, ins := range x.Body.List {
+				        if(ins.Pos() >= codePos) {
+					        continue;
+				        }
+				        switch insCasted := ins.(type) {
```

Comment:
You might be able to do this more simply with [ast.Walk](https://golang.org/pkg/go/ast/#Walk)

Reply:
Ok, thanks! I'll take a look at it.

Reply:
I rewritten the code to use ast.Walk, currently I am supporting only variable definitions (local for current function and global), I'll work on supporting types & function definitions etc.